project-base/storefront/components/Blocks/Login/LoginForm.tsx

```diff
@@ -0,0 +1,143 @@
+import { useLoginForm, useLoginFormMeta } from './loginFormMeta';
+import { ExtendedNextLink } from 'components/Basic/ExtendedNextLink/ExtendedNextLink';
+import { FacebookIcon } from 'components/Basic/Icon/FacebookIcon';
+import { GoogleIcon } from 'components/Basic/Icon/GoogleIcon';
+import { SeznamIcon } from 'components/Basic/Icon/SeznamIcon';
+import { WarningIcon } from 'components/Basic/Icon/WarningIcon';
+import { SubmitButton } from 'components/Forms/Button/SubmitButton';
+import { Form } from 'components/Forms/Form/Form';
+import { FormLine } from 'components/Forms/Lib/FormLine';
+import { PasswordInputControlled } from 'components/Forms/TextInput/PasswordInputControlled';
+import { TextInputControlled } from 'components/Forms/TextInput/TextInputControlled';
+import { useDomainConfig } from 'components/providers/DomainConfigProvider';
+import { TIDs } from 'cypress/tids';
+import { GtmMessageOriginType } from 'gtm/enums/GtmMessageOriginType';
+import useTranslation from 'next-translate/useTranslation';
+import { FormProvider, SubmitHandler } from 'react-hook-form';
+import { usePersistStore } from 'store/usePersistStore';
+import { LoginFormType } from 'types/form';
+import { useLogin } from 'utils/auth/useLogin';
+import { blurInput } from 'utils/forms/blurInput';
+import { handleFormErrors } from 'utils/forms/handleFormErrors';
+import { getInternationalizedStaticUrls } from 'utils/staticUrls/getInternationalizedStaticUrls';
+
+export type LoginFormProps = {
+    defaultEmail?: string;
+    shouldOverwriteCustomerUserCart?: boolean;
+};
+
+export const LoginForm: FC<LoginFormProps> = ({ defaultEmail, shouldOverwriteCustomerUserCart }) => {
+    const { t } = useTranslation();
+    const cartUuid = usePersistStore((store) => store.cartUuid);
+    const { url } = useDomainConfig();
+    const [resetPasswordUrl] = getInternationalizedStaticUrls(['/reset-password'], url);
+
+    const [formProviderMethods] = useLoginForm(defaultEmail);
+    const formMeta = useLoginFormMeta(formProviderMethods);
+    const login = useLogin();
+
+    const getSocialNetworkLoginUrl = (
+        socialNetwork: 'facebook' | 'google' | 'seznam',
```

Comment:
what about introducing an enum for the values? :thinking:

Reply:
is it really necessary to solve it now? 🤔 in next US it will be changed to take this configuration from API, what do you say?

Reply:
ok